src/main/java/org/openstreetmap/atlas/geography/atlas/dynamic/DynamicAtlas.java

```diff
@@ -188,6 +190,15 @@ public DynamicAtlasPolicy getPolicy()
         return this.expander.getPolicy();
     }
 
+    /**
+     * @return A copy of the {@link Shard} to {@link Atlas} {@link Map} populated by the underlying
+     *         {@link DynamicAtlasExpander}
+     */
+    public Map<Shard, Atlas> getShardToAtlasMap()
+    {
+        return new HashMap<>(this.expander.getLoadedShards());
```

Comment:
The expander will have a bunch of null Atlases here for all the shards it searched but could not find an Atlas for. Since this was internal it was fine, however exposing it this way, it would be nice to filter all the entries with null value first.

Reply:
Is there ever a case where users may want those `null` atlases though? For example doesn't that happen when `DynamicAtlas` tries to load a valid shard, but fails to find an atlas? Perhaps we don't want to erase that information by wiping out the null mappings. We can leave it to users to wipe the null mappings. Or maybe this is overthinking it? Your thoughts?

Reply:
So this method here has all the shards already. Would that be enough to know which ones we got and which ones we do not?
https://github.com/osmlab/atlas/blob/6.1.4/src/main/java/org/openstreetmap/atlas/geography/atlas/dynamic/DynamicAtlas.java#L195
I am really weary we would get unwanted NPEs here.

Reply:
Ah ok cool. As long as there is a way to get that info then we are good. I will make sure to note in the javadoc that the filtering is done for you.